### DISCUSSION TITLE
validation errors for Activity segment_efforts

### DISCUSSION THREAD

**Discussion by inspiran** [hozn/stravalib#451] [Q&A]
Hi, i am facing sync issues with Strava while retrieving an activity, seems to be just a Walk.

Before reporting it as a bug (pydantic) could one give me a hint how to debug (where to put breakpoints) to understand which segment type seems not to be supported?   

And maybe, is there a work around at this point?

The error:
4 validation errors for Activity segment_efforts -> 0 -> segment -> activity_type -> __root__ unexpected value; permitted: 'AlpineSki', ...

Thanks!

**Comment by inspiran** [comment]
It seems the segment activity type is WaterSport, is this something new?

**Reply by lwasser** [reply]
hey @inspiran can you please give us a code snippett that runs so we can try to recreate the issue and understand what is going on? it sounds like you're hitting the API and trying to fetch activities. You're seeing validation errors for specific activity types but is that when the activity is coming down initially? or when you are trying to look at specific segments within the activity? 

A reproducible code snippet would be super helpful.

**Comment by inspiran** [comment]
Hi, apologies for the late reply,  as it was something in production i can not just recreate the case.  
It happened when iterating over the results of get_activities( ) and as mentioned where the first segment was of type WaterSport.   Maybe it was a strava bug in the data (it was a walk, as far as i can tell you can not walk over water).

Note that i hav zero interest in segment efforts so I was also a bit surprised this part being parsed by pydantic.  I would expect a more on-demand kind of a logic.

**Comment by jsamoocha** [comment]
Hi, can it be that you're dealing with #441 ?

**Reply by inspiran** [reply]
Most likely yes

**Reply by jsamoocha** [reply]
Ok, that's a bug in Strava's API documentation (which is used to build our domain model). Let me see if I can create a workaround any time soon.
